macos/src/main/objectiveCpp/ThemeInfo.mm

```diff
@@ -0,0 +1,193 @@
+    if (getEnvStat == JNI_EDETACHED) {
+        detach = YES;
+        if (jvm->AttachCurrentThread((void **) &env, NULL) != 0) return;
+    } else if (getEnvStat == JNI_EVERSION) {
+        return;
+    }
+    jclass runnableClass = env->GetObjectClass(callback);
+    jmethodID runMethodId = env->GetMethodID(runnableClass, "run", "()V");
+    if (runMethodId) {
+        env->CallVoidMethod(callback, runMethodId);
+    }
+    if (env->ExceptionCheck()) {
+        env->ExceptionDescribe();
+    }
+    if (detach) jvm->DetachCurrentThread();
+}
+
+- (void)notificationEvent:(NSNotification *)notification {
+    [self runCallback];
+}
+
+@end
+
+JNIEXPORT jboolean JNICALL
+Java_com_github_weisj_darklaf_platform_macos_JNIThemeInfoMacOS_isDarkThemeEnabled(JNIEnv *env, jclass obj) {
+JNF_COCOA_ENTER(env);
+    if(@available(macOS 10.14, *)) {
+        NSString *interfaceStyle = [[NSUserDefaults standardUserDefaults] stringForKey:KEY_APPLE_INTERFACE_STYLE];
+        // interfaceStyle can be nil (light mode) or "Dark" (dark mode).
+        BOOL isDark = [VALUE_DARK caseInsensitiveCompare:interfaceStyle] == NSOrderedSame;
+        if (@available(macOS 10.15, *)) {
+            // Catalina
+            BOOL switchesAutomatically = [[NSUserDefaults standardUserDefaults] boolForKey:KEY_SWITCHES_AUTOMATICALLY];
+            if (switchesAutomatically) {
+                // If switchesAutomatically == YES the roles of "Dark" and nil are changed.
+                return (jboolean) !isDark;
+            }
+        }
+        // Mojave or switchesAutomatically == NO.
+        return (jboolean) isDark;
+    } else {
+        return (jboolean) NO;
+    }
+JNF_COCOA_EXIT(env);
+    return NO;
+}
+
+JNIEXPORT jboolean JNICALL
+Java_com_github_weisj_darklaf_platform_macos_JNIThemeInfoMacOS_isHighContrastEnabled(JNIEnv *env, jclass obj) {
+JNF_COCOA_ENTER(env);
+    return (jboolean) NSWorkspace.sharedWorkspace.accessibilityDisplayShouldIncreaseContrast;
+JNF_COCOA_EXIT(env);
+    return (jboolean) NO;
+}
+
+JNIEXPORT jint JNICALL
+Java_com_github_weisj_darklaf_platform_macos_JNIThemeInfoMacOS_nativeGetAccentColor(JNIEnv *env, jclass obj) {
+JNF_COCOA_ENTER(env);
+    if (@available(macOS 10.14, *)) {
+        BOOL hasAccentSet = ([[NSUserDefaults standardUserDefaults] objectForKey:KEY_ACCENT_COLOR] != nil);
+        if (hasAccentSet) {
+            return (jint) ([[NSUserDefaults standardUserDefaults] integerForKey:KEY_ACCENT_COLOR]);
+        }
+    }
+    return (jint) VALUE_DEFAULT_ACCENT_COLOR;
+JNF_COCOA_EXIT(env);
+    return (jint) VALUE_NO_ACCENT_COLOR;
+}
+
+JNIEXPORT jint JNICALL
+Java_com_github_weisj_darklaf_platform_macos_JNIThemeInfoMacOS_nativeGetSelectionColor(JNIEnv *env, jclass obj) {
+JNF_COCOA_ENTER(env);
+    NSColorSpace *rgbSpace = [NSColorSpace genericRGBColorSpace];
+    NSColor *accentColor = [[NSColorList colorListNamed: KEY_SYSTEM_COLOR_LIST] colorWithKey:KEY_SELECTION_COLOR];
```

Comment:
```
Non-Java exception raised, not handled! (Original problem: *** -redComponent not valid for the NSColor Catalog color: System selectedTextBackgroundColor; need to first convert colorspace.)
        at com.github.weisj.darklaf.platform.macos.JNIThemeInfoMacOS.nativeGetSelectionColor(Native Method)
```

Need to convert the color to the colorspace.

Reply:
Ups. Fixed

Reply:
```
-[NSDynamicSystemColor usingColorSpace:]: unrecognized selector sent to instance 0x7fa0fad70fa0
Exception in thread "AWT-EventQueue-0" java.lang.RuntimeException: Non-Java exception raised, not handled! (Original problem: -[NSDynamicSystemColor usingColorSpace:]: unrecognized selector sent to instance 0x7fa0fad70fa0)
        at com.github.weisj.darklaf.platform.macos.JNIThemeInfoMacOS.nativeGetSelectionColor(Native Method)
```